webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/quartz/SmartlingPullTranslationsJob.java

```diff
@@ -0,0 +1,82 @@
+package com.box.l10n.mojito.service.thirdparty.smartling.quartz;
+
+import com.box.l10n.mojito.quartz.QuartzJobInfo;
+import com.box.l10n.mojito.quartz.QuartzPollableJob;
+import com.box.l10n.mojito.quartz.QuartzPollableTaskScheduler;
+import com.box.l10n.mojito.service.repository.RepositoryRepository;
+import java.util.stream.LongStream;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class SmartlingPullTranslationsJob
+    extends QuartzPollableJob<SmartlingPullTranslationsJobInput, Void> {
+
+  @Autowired QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+
+  @Autowired RepositoryRepository repositoryRepository;
+
+  @Override
+  public Void call(SmartlingPullTranslationsJobInput input) throws Exception {
+
+    LongStream.range(0, batchesFor(input.getSingularCount(), input.getBatchSize()))
+        .forEach(
+            num -> {
+              SmartlingPullBatchFileJobInput smartlingPullBatchFileJobInput =
+                  new SmartlingPullBatchFileJobInput();
+              smartlingPullBatchFileJobInput.setBatchNumber(num);
+              smartlingPullBatchFileJobInput.setFilePrefix("singular");
+              setupInput(input, smartlingPullBatchFileJobInput);
+
+              QuartzJobInfo<SmartlingPullBatchFileJobInput, Void> quartzJobInfo =
+                  QuartzJobInfo.newBuilder(SmartlingPullBatchFileJob.class)
+                      .withInput(smartlingPullBatchFileJobInput)
+                      .withInlineInput(false)
+                      .withScheduler(input.getSchedulerName())
+                      .withParentId(getParentId())
+                      .build();
+              quartzPollableTaskScheduler.scheduleJob(quartzJobInfo);
+            });
+
+    LongStream.range(0, batchesFor(input.getPluralCount(), input.getBatchSize()))
+        .forEach(
+            num -> {
+              SmartlingPullBatchFileJobInput smartlingPullBatchFileJobInput =
+                  new SmartlingPullBatchFileJobInput();
+              smartlingPullBatchFileJobInput.setBatchNumber(num);
+              smartlingPullBatchFileJobInput.setFilePrefix("plural");
+              setupInput(input, smartlingPullBatchFileJobInput);
```

Comment:
some duplicated, maybe instantiate `SmartlingPullBatchFileJobInput` in the function and pass the file prefix as param?